### ISSUE TITLE
 Request to Add "console": "integratedTerminal" Functionality

### ISSUE THREAD

**Issue by jiandan720** [tomblind/local-lua-debugger-vscode#81]
Dear Tom,

I hope this email finds you well. I am writing to express my sincere gratitude for your invaluable contribution to the development community through your local-lua-debugger-vscode extension. As a beginner in Lua programming, I have found your plugin to be an indispensable tool for debugging my code within Visual Studio Code.

I have been using your extension extensively and have found it to be remarkably useful. The features and functionality it provides have greatly enhanced my coding experience and improved my productivity. The ability to debug Lua scripts directly within the VS Code environment has been incredibly helpful for me as a learner.

I would like to make a humble request for a feature enhancement in your extension. It would be immensely beneficial if you could add support for the "console": "integratedTerminal" option in the debugging configuration. As a beginner, I am still unfamiliar with performing source code modifications, and having this feature available would greatly simplify the debugging process for me. With integrated terminal support, I would be able to view the debugging output directly within the VS Code terminal, which would be incredibly convenient.

Once again, I want to express my heartfelt appreciation for your hard work and dedication in developing this extension. It has made a significant impact on my learning journey, and I am confident it has benefited many other Lua developers as well.

Thank you for considering my request. I understand that you may have numerous commitments and priorities, but I genuinely believe that this enhancement would greatly enhance the usability of your extension for beginners like myself.

I look forward to hearing from you and eagerly await any updates or developments in your future releases.

Warm regards,

**Comment by edzillion** [comment]
@jiandan720 It's an old post but FWIW this works for me:

```json
    {
      "name": "Busted",
      "type": "lua-local",
      "request": "launch",
      "program": {
        "command": "busted"
      },
      "cwd": "${workspaceFolder}",
      "args": [
        "${file}"
      ],
      "integratedTerminal": true ,
    }
```

and then add these lines to the top of your test file:
```lua
if os.getenv("LOCAL_LUA_DEBUGGER_VSCODE") == "1" then
  require("lldebugger").start()
end
```

**Comment by ogsadmin** [comment]
Hi @edzillion,

unfortunately this is not working for me - can you maybe provide a bit more information (which version you are using and how/where you access the console), as I am also interested in this feature...

I use V0.3.3 from the integrated addon list:
![image](https://github.com/tomblind/local-lua-debugger-vscode/assets/50247998/f36e0b94-ce3a-4e83-918f-8623e08e3432)

The element "integratedTerminal" is not recognized and starting a debug session does not show a new terminal also:
![image](https://github.com/tomblind/local-lua-debugger-vscode/assets/50247998/0c035fb5-7c38-44dd-83ad-6a2c4825b5e5)

Thanks!

**Comment by edzillion** [comment]
Yeah @ogsadmin  I get that warning too but it still works fine. Check in the `DEBUG CONSOLE` tab there and you _should_ see some output. 

I wrote up a more detailed post on this issue here: https://forum.exercism.org/t/how-to-debug-busted-unit-tests-in-vscode/8560

**Comment by ogsadmin** [comment]
Thanks @edzillion! This is actually working pretty well (if I select the right console window) - it even uses the correct scope (e.g. shows local variables). Unfortunately assignments don't work, so it's not a full repl:
![image](https://github.com/tomblind/local-lua-debugger-vscode/assets/50247998/d7264657-f0f9-43b6-969c-ec464092af66)

**Comment by trymeouteh** [comment]
Please add this feature to allow us to debug lua scripts that require the user input in the terminal. For example this script below will get stuck on line 2 since it is not possible to enter a prompt into the terminal when using the debugger.

```
io.write('Enter user input: ')
userInput = io.read()

print(userInput)
```
